tests/test_handler.rs

```diff
@@ -0,0 +1,44 @@
+struct CustomHandler {
+    msg: &'static str,
+}
+
+impl anyhow::ReportHandler for CustomHandler {
+    fn report(
+        &self,
+        _error: &(dyn std::error::Error + 'static),
+        f: &mut std::fmt::Formatter<'_>,
+    ) -> std::fmt::Result {
+        write!(f, "{}", self.msg)
+    }
+}
+
+#[test]
+fn test_custom_hook() {
+    let expected = "hook is set!";
+    anyhow::set_hook(Box::new(move |_error| {
+        Box::new(CustomHandler { msg: expected })
+    }))
+    .unwrap();
+
+    let report = anyhow::anyhow!("heres the message!");
+    let actual = format!("{:?}", report);
+
+    assert_eq!(expected, actual);
+}
+
+#[test]
+fn test_mutable_hook() {
+    let fake_expected = "hook is set!";
+    let real_expected = "the context was modified!";
+
+    anyhow::set_hook(Box::new(move |_error| {
+        Box::new(CustomHandler { msg: fake_expected })
+    }))
+    .unwrap();
+
+    let mut report = anyhow::anyhow!("heres the message!");
+    report.handler_mut().downcast_mut::<CustomHandler>().msg = real_expected;
```

Comment:
t seems that I misunderstood how using Any as an extension trait would work, so this may have to be implemented the same way downcasting on std::error::Error is, assuming that can even be done outside of std.